### PR TITLE
feat: set max response size for auth server on start

### LINKED ISSUE #3131: increase max response size for engine rpc handler
### Describe the feature

>  Failed to update deposit cache: GetDepositLogsFailed("eth_getLogs call failed ServerMessage { code: -32008, message: \"Response is too big\" }"), retry_millis: 60000, service: deposit_contract_rpc

we should increase memory settings for engine rpc server

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xJepsen**
Closes #3131

### CODE REVIEW

File: crates/rpc/rpc-builder/src/auth.rs [hunk 1]
```diff
@@ -24,6 +24,8 @@ use std::{
     net::{IpAddr, Ipv4Addr, SocketAddr},
     time::{Duration, SystemTime, UNIX_EPOCH},
 };
+/// Default max response size in MB.
+pub(crate) const RPC_DEFAULT_MAX_RESPONSE_SIZE_MB: u32 = 100;
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/auth.rs:28]: I don't want an additional variable for this,

I'd rather have another field for the authconfig type instead that we then set in cli args

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/rpc/rpc-builder/src/auth.rs (hunk 1 above)
Comment (0xJepsen) [reply crates/rpc/rpc-builder/src/auth.rs]: Sounds good i made the requested changes

File: crates/rpc/rpc-builder/src/auth.rs [hunk 2]
```diff
@@ -122,19 +122,21 @@ pub struct AuthServerConfig {
     pub(crate) socket_addr: SocketAddr,
     /// The secrete for the auth layer of the server.
     pub(crate) secret: JwtSecret,
+    /// The maximum response size for the server.
+    pub(crate) rpc_max_response_size: u32,
```
Comment (mattsse) [inline crates/rpc/rpc-builder/src/auth.rs:126]: I had another look at:

https://github.com/paradigmxyz/reth/blob/4d3d4974a48b4e838ccda50c3171da11c4885a13/crates/rpc/rpc-builder/src/lib.rs#L944-L946

and I'd like to replicate this for `AuthServerConfig`.

so basically:
1. a field `server_{config|builder}` 
2. then a separate function just like:

https://github.com/paradigmxyz/reth/blob/4d3d4974a48b4e838ccda50c3171da11c4885a13/crates/rpc/rpc-builder/src/lib.rs#L995-L1003

**Review by mattsse — CHANGES_REQUESTED** [review]
I have changed my mind about the "rpc_max_response_size" field 🙈
I would rather use the actual "ServerBuilder" as the field, as we already do in the regular Rpc builder. I think this is the better solution

**Comment by gakonst** [comment]
Really nice work here @0xJepsen! Are there any odds we can have this in by Monday? Else we would like to take it over I think, as we're trying to solve all the node operator QoL issues ASAP! Appreciate you taking this on.

**Comment by 0xJepsen** [comment]
> Really nice work here @0xJepsen! Are there any odds we can have this in by Monday? Else we would like to take it over I think, as we're trying to solve all the node operator QoL issues ASAP! Appreciate you taking this on.

I'm afraid im offline this weekend visiting family. Wish i could have gotten to it today. I could maybe work on in Sunday night. But i also wont be offended if you all take it over.

**Comment by gakonst** [comment]
Thank you sir! Will circle back Monday...enjoy the family time :)

**Review by mattsse — APPROVED** [review]
ty @0xJepsen, the only change I made was replacing the request limit arg with Serverbuilder

**Pull request merged**